### PR TITLE
Add the address model and view and tests!

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Issue Number

#246 

## Purpose/Implementation Notes

This adds the address model for users. It also adds an `address` field on the MaterialRequest model.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Functional tests

Unit Tests!!!!

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/resources_portal/test/views/test_address.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+from resources_portal.models import Address
+from resources_portal.test.factories import AddressFactory, UserFactory
+
+fake = Faker()
+
+
+class AddressListTestCase(APITestCase):
+    """
+    Tests /addresss list operations.
```
Comment (Philinphiladelphia) [inline api/resources_portal/test/views/test_address.py:16]:
```suggestion
    Tests /address list operations.
```

**Comment by kurtwheeler** [comment]
Ok I added some docs!!!

File: docs/api/address.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+{
+   "address_line_1": "11 Ranch Lane",
+   "address_line_2": "Suite 3000",
+   "country": "US",
+   "id": 4,
+   "institution": "Ranch Labs Inc.",
+   "locality": "Ranchville",
+   "name": "My Lab"s Address",
+   "postal_code": "12345",
```
Comment (Philinphiladelphia) [inline docs/api/address.md:46]: Slight issue with the quotation marks here.

File: docs/api/address.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+
+{
+   "address_line_1": "11 Ranch Lane",
+   "address_line_2": "Suite 3000",
+   "country": "US",
+   "id": 4,
+   "institution": "Ranch Labs Inc.",
+   "locality": "Ranchville",
+   "name": "My Lab"s Address",
```
Comment (Philinphiladelphia) [inline docs/api/address.md:78]: Same issue here.

File: docs/api/address.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+   "postal_code": "12345",
+   "saved_for_reuse": True,
```
Comment (Philinphiladelphia) [inline docs/api/address.md:80]:
```suggestion
   "saved_for_reuse": true,
```

File: docs/api/address.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+200 OK
+
+{
+   "address_line_1": "11 Ranch Lane",
+   "address_line_2": "Suite 3000",
+   "country": "US",
+   "id": 4,
+   "institution": "Ranch Labs Inc.",
+   "locality": "Ranchville",
+   "name": "My Lab"s Address",
```
Comment (Philinphiladelphia) [inline docs/api/address.md:133]: Once more here.

**Comment by kurtwheeler** [comment]
I addressed those!!

File: docs/api/address.md [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+{
+   "address_line_1": "11 Ranch Lane",
+   "address_line_2": "Suite 3000",
+   "country": "US",
+   "id": 4,
+   "institution": "Ranch Labs Inc.",
+   "locality": "Ranchville",
+   "name": "My Lab's Address",
+   "postal_code": "12345",
+   "saved_for_reuse": True,
```
Comment (Philinphiladelphia) [inline docs/api/address.md:47]:
```suggestion
   "saved_for_reuse": true,
```

File: docs/api/address.md [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
```
Comment (Philinphiladelphia) [inline docs/api/address.md:135]:
```suggestion
   "saved_for_reuse": true,
```

**Pull request merged**